src/components/course/EnrollButton.jsx

```diff
@@ -33,20 +35,20 @@ export default function EnrollButton() {
     pacingType,
     courseUuid,
   } = activeCourseRun;
-  const [isEnrollDisabled, setIsEnrollDisabled] = useState(false);
-  const [emailOptIn] = useState(false);
-  const [enrollmentSubmitted, setEnrollmentSubmitted] = useState(false);
 
-  /*
-   * TODO: this behavior mimics the existing B2C enrollment flow. we will want
-   * to redirect learners to the basket flow instead of the track selection page
-   */
-  useEffect(() => {
-    if (enrollmentSubmitted) {
-      // redirect to track selection page
-      window.location.href = `${process.env.LMS_BASE_URL}/course_modes/choose/${key}`;
-    }
-  }, [enrollmentSubmitted]);
+  // This enrollment URL assumes that the learner has access to the course through their subscription license, and they
+  // are using the license to enroll.
+  // TODO: Update to conditionally use the DSC flow with a license uuid only when the learner is actually using a
+  // license.
+  const enrollOptions = {
+    license_uuid: subscriptionLicense.uuid,
+    course_id: key,
+    enterprise_customer_uuid: enterpriseConfig.uuid,
+    next: `${process.env.LMS_BASE_URL}/dashboard`,
+    failure_url: global.location,
+  };
+  const licenseEnrollmentUrl = `${process.env.LMS_BASE_URL}/enterprise/grant_data_sharing_permissions/?${qs.stringify(enrollOptions)}`;
```

Comment:
I couldn't fully test the entire flow of DSC locally as there seemed to be some incorrect configuration in the demo course. The failure and next steps worked correctly, but I was not able to enroll (due to what seemed like config issues). I will test this more on stage where configuration is hopefully better and adjust accordingly